### DISCUSSION TITLE
How to understand the channel of a trigger or a digital output event?

### DISCUSSION THREAD

**Discussion by zhengliuX** [pulseq/pulseq#56] [Q&A] [answered]
Dear all,
    Comments here say we are creating a trigger or a digital output event based on a given channel. So if I do `mr.makeTrigger('physio1')`, how do I know if this is an ECG trigger, or a Pulse trigger?
  
![image](https://github.com/user-attachments/assets/e76cbd35-750f-4a91-befe-dae74175aced)

![image](https://github.com/user-attachments/assets/d95cc665-3900-42ad-afa4-94a423884630)

Best,
Zheng

**Comment by QingpingChen1201** [comment, marked as answer]
Hi Zheng,
The ECG trigger works as below:
1) you create a trigger object like: 
trig=mr.makeTrigger('physio1','duration', 2000e-6); % duration after 
2) you add it to the timing of your sequence like: 
seq.addBlock(trig); % wait for the cardiac trigger
This will cause the sequence to stop at the beginning of the block and wait for the trigger to happen. The physiological trigger signal to use needs to be manually selected on the user interface of the Pulseq interpreter sequence.
The example for using this feature can be found in writeCineGradientEcho.m under the following path: https://github.com/pulseq/pulseq/blob/master/matlab/demoSeq/writeCineGradientEcho.m

The pulse output trigger is created through a different function: makeDigitalOutputTrigger(channel, ...) with different channels (possible channels are 'ext1', 'osc0'... in Siemens). For example:
% define the output trigger to play out with every slice excitatuion
trig=mr.makeDigitalOutputPulse('osc0','duration', 100e-6); % possible channels: 'osc0','osc1','ext1'
seq.addBlock(trig) ; % inset the trigger
This digital output trigger sends out a signal through the given channel for you to detect but does not stop the sequence.
If you need more information, please feel free to let us know.
Hope this helps!

Best regards,
Qingping and the Pulseq team

**Reply by zhengliuX** [reply]
Thanks a lot! Really helps!

**Reply by mkulam** [reply]
What does physio1 and physio2 mean in makeTrigger function? If I want to use PPG trigger, what should I use?

**Reply by m-a-x-i-m-z** [reply]
Sorry for my ignorance, but what is *PPG trigger*? 
--M.

**Reply by mkulam** [reply]
Photoplethysmography or pulse oximeter. Trigger based on pulse oximeter reading from the finger.

**Reply by m-a-x-i-m-z** [reply]
I assume *physio1* is a better starting point. If it doesn't work, try the other one. And, please share your experiences hwith us here :-)
--M.

**Reply by mkulam** [reply]
Ok a small update. I looked at the pulseq source code, the trigger was set to ALL. So any of the physiological signals should work. I tried physio1 again with pulse trigger option. It worked this time around. I am not sure exactly what changed. Last time the scanner was somehow assuming the period of the pulse wave even though I did not wear the sensor.
